File: Data Structures & Algorithms/subtree-of-a-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:   
    def isSubtree(self, root: Optional[TreeNode], subRoot: Optional[TreeNode]) -> bool:

        def isSym(a, b):
            if not a and not b: return True
            if not a or not b:
                return False
            return a.val == b.val and isSym(a.left, b.left) and isSym(a.right, b.right)
        
        def dfs(node):
            if not node:
                return False
            if node.val == subRoot.val and isSym(node, subRoot):
                return True
            else:
                return (dfs(node.left) or
                dfs(node.right))
        
            return  False
        return dfs(root)
        
            

File: Data Structures & Algorithms/range-sum-of-bst/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int rangeSumBST(TreeNode root, int low, int high) {
        return bfs(root, low, high);
    }

    private int bfs(TreeNode curr, int low, int high) {
        if (curr == null) return 0;
        if (curr.val >= low && curr.val <= high) {
            return curr.val + bfs(curr.left, low, high) + bfs(curr.right, low, high);
        }
        return 0;
    }
}